multiapps-controller-client/src/main/java/org/cloudfoundry/multiapps/controller/client/util/ResilientCloudOperationExecutor.java

```diff
@@ -38,6 +48,27 @@ public ResilientCloudOperationExecutor withStatusesToIgnore(HttpStatus... status
         return this;
     }
 
+    public <T> T executeWithExponentialBackoff(Function<Duration, T> operation) {
+        int waitTimeBetweenRetriesInMillis = DEFAULT_TIMEOUT_RETRY_WAIT_TIME_IN_MILLIS;
+        int retryIndex = 1;
+        for (int i = 1; i < RETRY_COUNT_RESPONSE_TIME_BACKOFF.size(); i++) {
+            try {
+                return operation.apply(RETRY_COUNT_RESPONSE_TIME_BACKOFF.get(retryIndex++));
+            } catch (RuntimeException e) {
+                handle(e);
+                if (e.getCause() instanceof java.util.concurrent.TimeoutException
+                    || e instanceof io.netty.handler.timeout.TimeoutException) {
+                    LOGGER.info("Waiting: {} ms before retrying with timeout of: {} ms", waitTimeBetweenRetriesInMillis,
```

Comment:
maybe extract message in an external constant. but either way is ok

Reply:
extracted